packages/rlc-common/src/buildSamples.ts

```diff
@@ -0,0 +1,835 @@
+  PathMetadata,
+  OperationParameter,
+  ObjectSchema,
+  SchemaContext,
+  Schema,
+  ArraySchema,
+  DictionarySchema
+} from "./interfaces.js";
+import { sampleTemplate } from "./static/sampleTemplate.js";
+// @ts-ignore: to fix the handlebars issue
+import hbs from "handlebars";
+import * as path from "path";
+import { isObjectSchema } from "./helpers/schemaHelpers.js";
+import { getImmediateParentsNames } from "./buildObjectTypes.js";
+
+export function buildSamples(model: RLCModel) {
+  if (!model.options || !model.options.packageDetails) {
+    return;
+  }
+  const sampleGroups: RLCSampleGroup[] | undefined = model.sampleGroups;
+  if (!sampleGroups || sampleGroups.length === 0) {
+    return;
+  }
+  const sampleFiles: RLCFile[] = [];
+  for (const sampleGroup of sampleGroups) {
+    const sampleGroupFileContents = hbs.compile(sampleTemplate, {
+      noEscape: true
+    });
+    const filePath = path.join("samples-dev", `${sampleGroup.filename}.ts`);
+    sampleFiles.push({
+      path: filePath,
+      content: sampleGroupFileContents(sampleGroup)
+    });
+  }
+  return sampleFiles;
+}
+
+export function buildSamplesOnMockContent(model: RLCModel) {
+  if (model.options?.multiClient || model.options?.isModularLibrary) {
+    // Not support to generate if multiple clients
+    // Not support to generate if modular libraries
+    return;
+  }
+  const rlcSampleGroups: RLCSampleGroup[] = [];
+  // Get all paths
+  const paths: Paths = model.paths;
+  const clientName = getClientName(model);
+  const clientInterfaceName = clientName.endsWith("Client")
+    ? `${clientName}`
+    : `${clientName}Client`;
+  const defaultFactoryName = normalizeName(
+    camelCase(`create ${clientInterfaceName}`),
+    NameType.Method
+  );
+  const packageName = model.options?.packageDetails?.name ?? "";
+  const methodParameterMap = buildMethodParamMap(model);
+  const schemaObjectMap = buildSchemaObjectMap(model);
+  for (const path in paths) {
+    const pathDetails = paths[path];
+    const methods = pathDetails.methods;
+    for (const method in methods) {
+      const importedDict: Record<string, Set<string>> = {};
+      const detail: OperationMethod = methods[method][0];
+      const operatonConcante = getOperationConcate(
+        detail.operationName,
+        pathDetails.operationGroupName,
+        model.options?.sourceFrom
+      );
+      const operationPrefix = normalizeName(
+        camelCase(transformSpecialLetterToSpace(operatonConcante)),
+        NameType.Method
+      );
+      const sampleGroup: RLCSampleGroup = {
+        filename: `${operationPrefix}Sample`,
+        defaultFactoryName,
+        clientPackageName: packageName,
+        samples: []
+      };
+
+      // initialize the sample
+      const sample: RLCSampleDetail = {
+        description: `call operation ${detail.operationName}`,
+        name: `${operationPrefix}Sample`,
+        path,
+        defaultFactoryName,
+        clientParamAssignments: [],
```

Comment:
As many values are empty, can we just skip the initialization for them?

Reply:
Empty values are meaningful to sample template if not set as empty it would be `undefined`.

I prefer to have a centralized place to set the default value, otherwise the relevant setting method needs to handle them one by one.